src/harness/testcases/WINNF_FT_S_FDB_testcase.py

```diff
@@ -0,0 +1,1240 @@
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam'],
+        'measCapability': device_b['measCapability']
+    }
+    del device_b['cbsdCategory']
+    del device_b['airInterface']
+    del device_b['installationParam']
+    del device_b['measCapability']
+
+    conditionals = [conditionals_b]
+
+    # Create the actual config.
+    config = {
+        'registrationRequests': [device_a, device_b],
+        'grantRequests': [
+            [grant_g1_a, grant_g1_b],
+            [grant_g2_a, grant_g2_b]],
+        'conditionalRegistrationData': conditionals
+        # TODO
+        # Need to add data base configurations
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_FDB_4_default_config)
+  def test_WINNF_FT_S_FDB_4(self, config_filename):
+    """FSS Database Update: FSS Site Modification.
+
+    Check SAS UUT responds with HeartbeatResponse either SUCCESS or
+    SUSPENDED_GRANT for CBSD C's location X near FSS site before and
+    after FSS database record modification.
+    """
+
+    # Load the configuration file
+    config = loadConfig(config_filename)
+
+    # Step 1: Register device(s) 'C' (at location 'X') with conditional registration data
+    cbsd_ids = self.assertRegistered(config['registrationRequests'],
+                                     config['conditionalRegistrationData'])
+
+    # TODO
+    # Step 2: Create FSS database which includes atleast one FSS site near location 'X'.
+
+    # Step 3: Trigger daily activities
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Step 4: Send Grant Request 'G1'
+    # Forming of 'G1' grants for all the configured grants
+    addCbsdIdsToRequests(cbsd_ids, config['grantRequests'][0])
+    grant_request_g1 = {'grantRequest': config['grantRequests'][0]}
+
+    # Send grant request 'G1'
+    grant_response_g1 = self._sas.Grant(grant_request_g1)['grantResponse']
+
+    # Check grant 'G1' response (responseCode should be SUCCESS(0))
+    grant_ids = []
+    for resp in grant_response_g1:
+      self.assertEqual(resp['response']['responseCode'], 0)
+      grant_ids.append(resp['grantId'])
+
+    del grant_request_g1, grant_response_g1
+
+    # Step 5.1: Sending Heartbeat Request
+    # Construct heartbeat message
+    heartbeat_requests = {'heartbeatRequest': []}
+    for cbsd_id, grant_id in zip(cbsd_ids, grant_ids):
+      heartbeat_requests['heartbeatRequest'].append({
+        'cbsdId': cbsd_id,
+        'grantId': grant_id,
+        'operationState': 'GRANTED'
+      })
+    heartbeat_responses = self._sas.Heartbeat(heartbeat_requests)['heartbeatResponse']
+
+    # Check the heartbeat response code is either 0(SUCCESS) or
+    # 501(SUSPENDED_GRANT)
+    for resp in heartbeat_responses:
+      self.assertIn(resp['response']['responseCode'], [0, 501])
```

Comment:
This should be configurable: the tester will specify a set of valid response codes for each heartbeat response.

Reply:
FDB4.     Used the expectedResponseCodes from the config file during checking   heartbeat response.